modules/pam_userdb/pam_userdb.c

```diff
@@ -129,6 +131,76 @@ _pam_parse (pam_handle_t *pamh, int argc, const char **argv,
   return ctrl;
 }
 
+/*
+ * Database abstraction functions
+ */
+static void *
+db_open(const char *database, mode_t file_mode)
+{
+    void *dbm;
+
+#ifdef HAVE_GDBM_H
+    dbm = gdbm_open(database, 4096, GDBM_READER, file_mode, NULL);
+#else
+    dbm = dbm_open(database, O_RDONLY, file_mode);
+#endif /* HAVE_GDBM_H */
+
+    return dbm;
+}
+
+static datum
+db_firstkey(void *dbm)
+{
+    datum key;
+
+#ifdef HAVE_GDBM_H
+    key = gdbm_firstkey(dbm);
+#else
+    key = dbm_firstkey(dbm);
+#endif /* HAVE_GDBM_H */
+
+    return key;
+}
+
+static datum
+db_nextkey(void *dbm, datum key UNUSED)
+{
+    datum nextkey;
+
+#ifdef HAVE_GDBM_H
+    nextkey = gdbm_nextkey(dbm, key);
```

Comment:
Sorry, how can `key` be `UNUSED` if it is passed down to `gdbm_nextkey`?

Reply:
It's unused for the non-gdbm option, where `key` isn't used. If I wouldn't have set the `UNUSED` keyword, then the CI would have failed. However, I'm open to other solutions.

Reply:
This means that the use of `UNUSED` keyword also has to be conditional.

Reply:
Done